File: meal_plan_api/models/mealModel.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

// This is the catalog of all possible meals a user can swap to.
const MealSchema = new Schema({
  name: { type: String, required: true },
  description: String,
  calories: Number,
  protein: Number,
  carbs: Number,
  fat: Number,
  imageUrl: { type: String, required: true },
});

module.exports = mongoose.model('Meal', MealSchema);

File: meal_plan_app/lib/models/meal_model.dart
class Meal {
  final String id;
  final String name;
  final String description;
  final String imageUrl;

  Meal({
    required this.id,
    required this.name,
    required this.description,
    required this.imageUrl,
  });

  factory Meal.fromJson(Map<String, dynamic> json) {
    return Meal(
      id: json['_id'],
      name: json['name'],
      description: json['description'] ?? 'No description available.',
      imageUrl: json['imageUrl'],
    );
  }
}